projects/core/components/svg/svg.component.ts

```diff
@@ -112,11 +126,11 @@ export class TuiSvgComponent {
     }
 
     private get isUrl(): boolean {
-        return this.icon.endsWith(`.svg`);
+        return typeof this.icon === `string` ? this.icon.endsWith(`.svg`) : false;
     }
 
     private get isSrc(): boolean {
-        return tuiIsPresumedHTMLString(this.icon);
+        return typeof this.icon === `string` ? tuiIsPresumedHTMLString(this.icon) : false;
```

Comment:
`return typeof this.icon === 'string' && tuiIsPresumedHTMLString(this.icon)`

Reply:
fixed